test/unit/manipulation.js

```diff
@@ -3099,3 +3099,71 @@ testIframe(
 		} );
 	}
 );
+
+// Custom event handler test case
+
+QUnit.test( "should handle custom '_se-custom-destroy' event correctly", function( assert ) {
+
+	// Set up the HTML structure using innerHTML within #qunit-fixture
+
+	assert.expect( 2 );
+
+	var fixture = jQuery( "#qunit-fixture" );
+
+	fixture.html( `
+    <div id="container">
+      <div class="guarded removeself" data-elt="one">
+        Guarded 1
+      </div>
+      <div class="guarded" data-elt="two">
+        Guarded 2
+      </div>
+      <div class="guarded" data-elt="three">
+        Guarded 3
+      </div>
+    </div>
+  ` );
+
+	// JavaScript code to be tested
+
+  jQuery.event.special[ "_se-custom-destroy" ] = {
+    remove: function( _handleObj ) {
+		var $t = jQuery( this );
+		console.log( $t.data( "elt" ) );
+		if ( $t.is( ".removeself" ) ) {
+			$t.remove();
+			}
+		}
+	};
+
+	// Attach the custom event handler
+
+	jQuery( ".guarded" ).on( "_se-custom-destroy", function( ) { } );
+
+	// Spy on console.log
+
+	var consoleLog = console.log;
+
+	var logMessages = [];
+
+	console.log = function( message ) {
+	logMessages.push( message );
+	};
+
+	// Trigger the event by emptying the container
+
+	jQuery( "#container" ).empty( );
+
+	// Verify that the elements with class 'removeself' were removed
+
+	assert.equal( jQuery( ".removeself" ).length, 0, "Elements with class 'removeself' should be removed" );
```

Comment:
This doesn't tell us much as we're removing almost everything.

Instead, I'd add one `assert.step` to the contents of the `remove` function and then verify those were called.